src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -151,59 +159,73 @@ private async Task<bool> IsReviewSame(ReviewModel review, RenderedCodeFile rende
-            }
-                
-
-            if(pullRequestModel.ReviewId != null)
-            {
-                // If baseline review was already created and if APIs in current commit doesn't match any of the revisions in generated review then create new baseline using main branch and compare again.
-                // If APIs are still different, find the diff against latest baseline.
-                review = await GetBaseLineReview(codeFile.Language, codeFile.PackageName, pullRequestModel, true);
-                review.ReviewId = pullRequestModel.ReviewId;
-                if (await IsReviewSame(review, renderedCodeFile))
+                // If base line is not available (possible if package is new or request coming from SDK automation)
+                review = new ReviewModel()
+                {
+                    Author = pullRequestModel.Author,
+                    CreationDate = DateTime.Now,
+                    Name = pullRequestModel.PackageName,
+                    IsClosed = false,
+                    FilterType = ReviewType.PullRequest,
+                    ReviewId = IdHelper.GenerateId()
+                };
+                var reviewUrl = reviewUrlPrefix + review.ReviewId;
+                stringBuilder.Append($"API review is created for `{codeFile.PackageName}`. You can review APIs [here]({reviewUrl}).").Append(Environment.NewLine);
+            }
+            else
+            {
+                // Check if API surface level matches with any revisions
+                var renderedCodeFile = new RenderedCodeFile(codeFile);
+                if (await IsReviewSame(review, renderedCodeFile))
                 {
-                    // We will run into this if some one makes unintended API changes in a PR and then reverts it back.
-                    // We must clear previous comment and update it to show no changes found.
+                    //Do not update the comment if review was already created and it matches with current revision.
+                    if (pullRequestModel.ReviewId != null)
+                        return "";
+
+                    //Baseline review was not created earlier or this is the first commit of PR
                     stringBuilder.Append($"API changes are not detected in this pull request for `{codeFile.PackageName}`");
                     return stringBuilder.ToString();
-                }
+                }
+
+                if (pullRequestModel.ReviewId != null)
+                {
+                    // If baseline review was already created and if APIs in current commit doesn't match any of the revisions in generated review then create new baseline using main branch and compare again.
+                    // If APIs are still different, find the diff against latest baseline.
+                    review = await GetBaseLineReview(codeFile.Language, codeFile.PackageName, pullRequestModel, true);
+                    if (review != null)

```

Comment:
Doesn't look like this can actually be null after the changes to GetBaseLineReview.

Reply:
It can still be null if PR submitted is for a new package( so no automatic review is present in that case). So we will run into this code only if developer makes additional commit with additional API changes. Automatic review is not available at that point to return as baseline.

Reply:
Won't it nullref then here - https://github.com/Azure/azure-sdk-tools/pull/2864/files#diff-cd7775ceaaca24a579fca2f459e2d216bfc15075b4bffdb75d9a943d92fbde7cR281

Reply:
Resolved. I have updated baseline method in a way to avoid returning null for review and it is already handled for the only case it returns null when a new PR is submitted for a new package by creating new review object right after the first call to BaseLine.